test/req/steps_test.exs

```diff
@@ -534,6 +534,18 @@ defmodule Req.StepsTest do
       assert resp.body == "foo"
     end
 
+    test "gzip error", c do
+      Bypass.expect(c.bypass, "GET", "/", fn conn ->
+        conn
+        |> Plug.Conn.put_resp_header("content-encoding", "x-gzip")
+        |> Plug.Conn.send_resp(200, "bad")
+      end)
+
+      assert_raise Req.DecompressError, "decompression failed with format 'gzip'", fn ->
+        Req.get!(c.url)
```

Comment:
nitpick: I'd like to use :plug feature as much as possible going forward, for everything that doesn't rely on network or some HTTP implementation details:

```suggestion
    test "gzip error" do
      plug = fn conn ->
        conn
        |> Plug.Conn.put_resp_header("content-encoding", "x-gzip")
        |> Plug.Conn.send_resp(200, "bad")
      end

      assert_raise Req.DecompressError, "decompression failed with format 'gzip'", fn ->
        Req.get!(plug: plug)
```